jumanji/environments/combinatorial/binpack/env.py

```diff
@@ -348,13 +355,13 @@ def step(self, state: State, action: Action) -> Tuple[State, TimeStep[Observatio
 
         return next_state, timestep
 
-    def render(self, state: State) -> None:
+    def render(self, state: State) -> Optional[jnp.ndarray]:
```

Comment:
Good afternoon, I do not think you should narrow the type to `jnp.ndarray` because this cannot handle 3D representations like {'objectId': {'vertices': [...], 'faces': [...], face_colours: [...], etc....}. Have you seen how [Brax](https://github.com/google/brax) does it? I think what they are done is great.

In summary, whilst yes you might want 2D views of the environment, we should also support visualising 3D environments.

Reply:
Good afternoon :slightly_smiling_face:

* Good point for 3d environment ! What would you suggest as return type then ? `Any` ?
* From what I understand in brax, they can either render an html viewer though brax.io.html.render or return an image through brax.io.image.render. The html.render uses some mesh under the hood, but does not return them. I think the mesh representation they use is anyway tightly coupled to their javascript viewer, so it would not make sens to export outside of brax. The interface is similar to what we have now: for a 3d environment we would represent it in whatever visualization framework we have (matplotlib here for binpack), but if we want to export something it would be a 2d image
* We could add a new mode like `RenderMode.MESH`, which would return a mesh representation, but it would have to be a standard mesh type usable by other libraries, do you know if such exists ?

Reply:
Apologies, I misread the PR and thought you wanted to enforce this return type on the Env interface, I am discarding this comment.